### PR TITLE
Add Modal

### PR DESCRIPTION
**Pull request by mickitron**

### CODE REVIEW

File: source/layouts/Navigation/index.js [hunk 1]
```diff
@@ -40,11 +57,40 @@ export default class extends Component {
             </div>
           </div>
           <div className={styles.ctaMenu}>
-            <Link to={process.env.LOGIN_LINK}>Login</Link>
+            <Button theme='link' onClick={() => (this.showModal())}>Login</Button>
```
Comment (lukebrooker) [inline source/layouts/Navigation/index.js]: This should really just be `onClick={this.showModal}`. Just need to make sure to bind `this` in the constructor.

File: source/layouts/Navigation/index.js [hunk 2]
```diff
@@ -40,11 +57,40 @@ export default class extends Component {
...
             <Link to={process.env.SIGNUP_LINK}>Signup</Link>
             <Button link to='#donate' theme='primaryOutline'>Donate</Button>
           </div>
         </div>
+        <Modal
+          className={styles.modal}
+          show={this.state.modalVisible}
+          onClose={this.closeModal.bind(this)}
```
Comment (lukebrooker) [inline source/layouts/Navigation/index.js]: Same type of thing here. `this.closeModal` and then bind in the constructor.

File: source/components/Button/styles.css [hunk 3]
```diff
@@ -29,8 +29,14 @@
...
 .primaryCaps {
   composes: primary;
-  text-transform: uppercase;
-  font-weight: 700;
+  composes: bold uppercase from '../../traits/type.css';
+}
+
+.primaryLogin {
+  composes: primary;
+  width: 240px;
```
Comment (lukebrooker) [inline source/components/Button/styles.css]: Fixed width is still here?

File: source/layouts/Navigation/index.js (hunk 1 above)
Comment (bradparker) [inline source/layouts/Navigation/index.js]: There's a couple of ways to skin this cat. Binding in the constructor is one, I think it's the most efficient one, it's also the most verbose.

This one creates a new function on each render:
```
<Button theme='link' onClick={() => this.showModal()}>
```

As does this one:
```
<Button theme='link' onClick={this.showModal.bind(this)}>
```

This one only creates a new function when the component is initialised.
```
// ...
constructor () {
  this.showModal = (e) => { /* horay */ }
}
// ...
<Button theme='link' onClick={this.showModal}>
```

Each function literal makes a new function, each call to `bind` makes a new function.

Handy things to know I think.

File: source/layouts/Navigation/index.js [hunk 4]
```diff
@@ -40,11 +59,40 @@ export default class extends Component {
...
           <div className={styles.ctaMenu}>
-            <Link to={process.env.LOGIN_LINK}>Login</Link>
+            <Button theme='link' onClick={this.showModal}>Login</Button>
             <Link to={process.env.SIGNUP_LINK}>Signup</Link>
             <Button link to='#donate' theme='primaryOutline'>Donate</Button>
           </div>
         </div>
+        <Modal
+          className={styles.modal}
+          show={this.state.modalVisible}
```
Comment (bradparker) [inline source/layouts/Navigation/index.js]: Add to the de-structuring above?

File: source/layouts/Navigation/index.js [hunk 5]
```diff
@@ -2,20 +2,38 @@ import React, { Component } from 'react'
 import styles from './styles.css'
 import Link from '../../components/Link'
 import Button from '../../components/Button'
+import Modal, {closeStyle} from 'simple-react-modal'
```
Comment (bradparker) [inline source/layouts/Navigation/index.js]: `closeStyle` appears to be unused.

File: source/layouts/Navigation/index.js (hunk 1 above)
Comment (lukebrooker) [inline source/layouts/Navigation/index.js]: Yep. That's why I mentioned it. Probably should of explained my reasoning more.

**Pull request merged**